src/dbnode/client/types.go

```diff
@@ -78,6 +78,10 @@ type Session interface {
 	// FetchTaggedIDs resolves the provided query to known IDs.
 	FetchTaggedIDs(namespace ident.ID, q index.Query, opts index.QueryOptions) (iter TaggedIDsIterator, exhaustive bool, err error)
 
+	// FetchTags resolves the optionally provided query to any known tag matchers.
+	// NB: this is a provisional function until the `session` contract is fully defined
+	FetchTags(namespace ident.ID, q index.Query, opts index.AggregateQueryOptions) (tags FetchedTags, exhaustive bool, err error)
```

Comment:
How would you feel about `Aggregate` instead?

Reply:
also, re: the return type - would be better to use an Iterator - allows us to avoid a bunch of allocs

Reply:
Sounds good, I like the iterator you added in #1483; re: `Aggregate` where do you think we change the context of the function from being a generic aggregation vs a tag completion; would that be entirely in query? Maybe meet halfway with `AggregateTags` 😛 ? We've already got tag specific stuff in this interface so wouldn't be a huge change, right?